### DISCUSSION TITLE
Manually move job to another queue?

### DISCUSSION THREAD

**Discussion by duncanb-q** [taskforcesh/bullmq#578] [Q&A]
Our jobs can be quite long running (minutes, or even hours).  We want the ability to move a long-running job to another queue where it will be processed by a 'lower-pri' pool of workers.  

Is this possible? 

```
const worker = new Worker("high-pri", async (job: Job) => {

    if (jobRunningTooLong) { 
        moveToQueue("low-pri", job);
    }
    return delayed; 
});

**Comment by manast** [comment]
I would suggest that you just add a new job to the other queue and return from this one.

**Reply by duncanb-q** [reply]
Thanks @manast 

yes, we can do that and that is likely the workaround we will go with.  The issue is that this messes with our metrics (we measure the time from scheduling a job to the time that the job is finished processing).
